### PR TITLE
SPSH-1137

### PR DESCRIPTION
**Pull request by he-meyer**
# Description

## Links to Tickets or other pull requests

https://ticketsystem.dbildungscloud.de/browse/SPSH-1137

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: src/modules/cron/cron.controller.ts [hunk 1]
```diff
@@ -171,4 +176,26 @@ export class CronController {
...
+    @ApiInternalServerErrorResponse({
+        description: 'Internal server error while trying to update VIDIS offers.',
+    })
+    public async updateServiceProvidersForVidisOffers(@Permissions() permissions: PersonPermissions): Promise<void> {
+        const hasCronJobPermission: boolean = await permissions.hasSystemrechteAtRootOrganisation([
+            RollenSystemRecht.CRON_DURCHFUEHREN,
+        ]);
+        if (!hasCronJobPermission) {
+            throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
+                SchulConnexErrorMapper.mapDomainErrorToSchulConnexError(new CronJobError('Ungenügende Berechtigungen')),
```
Comment (tikle-cap) [inline src/modules/cron/cron.controller.ts:196]: ich würde wahrscheinlich eher MissingPermissionsError mit einer englischen Fehlermeldung verwenden

File: src/modules/service-provider/domain/service-provider.service.ts [hunk 2]
```diff
@@ -23,4 +35,119 @@ export class ServiceProviderService {
 
         return Array.from(serviceProviders.values());
     }
+
+    public async updateServiceProvidersForVidis(): Promise<void> {
+        this.logger.info('Update of service providers for VIDIS offers triggered.');
+
+        const vidisKeycloakGroup: string = 'VIDIS-service';
```
Comment (tikle-cap) [inline src/modules/service-provider/domain/service-provider.service.ts:42]: soll das so hard coded sein?

File: src/modules/service-provider/domain/service-provider.service.ts [hunk 3]
```diff
@@ -23,4 +35,119 @@ export class ServiceProviderService {
...
+                if (existingServiceProvider.ok) {
+                    serviceProvider = ServiceProvider.construct(
+                        existingServiceProvider.value.id,
+                        existingServiceProvider.value.createdAt,
+                        existingServiceProvider.value.updatedAt,
+                        offer.offerTitle,
+                        ServiceProviderTarget.URL,
+                        offer.offerLink,
+                        ServiceProviderKategorie.ANGEBOTE,
+                        'ffb411f6-8287-4939-b87d-13b6a4e7fdfa', // This is a random UUID
```
Comment (tikle-cap) [inline src/modules/service-provider/domain/service-provider.service.ts:71]: Wofür ist die ID?

File: src/modules/cron/cron.controller.ts (hunk 1 above)
Comment (DeFi3298) [reply src/modules/cron/cron.controller.ts:196]: @he-meyer siehe person.controller.ts
image

**Review by DeFi3298 — CHANGES_REQUESTED** [review]

**Review by tikle-cap — APPROVED** [review]

**Review by DeFi3298 — APPROVED** [review]

File: src/modules/service-provider/domain/service-provider.service.ts [hunk 4]
```diff
@@ -23,4 +43,118 @@ export class ServiceProviderService {
...
+                        ServiceProviderSystem.NONE,
+                        false,
+                    );
+                    this.logger.info(`ServiceProvider for VIDIS offer '${serviceProvider.name}' was created.`);
+                }
+                const persistedServiceProvider: ServiceProvider<true> =
+                    await this.serviceProviderRepo.save(serviceProvider);
+                await Promise.allSettled(
+                    offer.schoolActivations.map(async (schoolActivation: string) => {
+                        const school: Result<
```
Comment (YoussefBouch) [inline src/modules/service-provider/domain/service-provider.service.ts:107]:
```suggestion
                        const orga: Result<
```

File: src/modules/service-provider/domain/service-provider.service.ts [hunk 5]
```diff
@@ -23,4 +42,117 @@ export class ServiceProviderService {
...
+                        ServiceProviderKategorie.ANGEBOTE,
+                        schulstrukturknoten,
+                        Buffer.from(angebot.angebotLogo, 'base64'),
+                        angebotLogoMediaType,
+                        vidisKeycloakGroup,
+                        vidisKeycloakRole,
+                        ServiceProviderSystem.NONE,
+                        false,
+                    );
+                    this.logger.info(`ServiceProvider for VIDIS Angebot '${serviceProvider.name}' already exists.`);
```
Comment (YoussefBouch) [inline src/modules/service-provider/domain/service-provider.service.ts:85]: Why do we need to construct the service provider when it already exists in the DB?

**Review by YoussefBouch — CHANGES_REQUESTED** [review]

**Review by YoussefBouch — APPROVED** [review]

**Pull request merged**